src/ui/transforms.js

```diff
@@ -40,40 +40,42 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
     };
 
     /* A transform to turn an array into a delimited string
-     * Values can be optionally be
-     * included only if they are string value and are truthy (i.e. not an empty
-     * string). Values can also be accessed via a specific path relative to each
-     * term.
-     * - separator: the string delimiter to be inserted between each term
-     * - stringOnly (optional): flag to exclude all falsy or non-string values
-     * - path (optional): an EL path on each item in the terms collection
+     * Values can also be accessed via a specific object path relative to each term.
+     * - "separator" (optional): the delimiter to be inserted between each term
+     * - "trailingSeparator" (optional): flag to include the separator at the end
+     * - "stringOnly" (optional): flag to allow only non-empty strings
+     * - "path" (optional): an EL path on each item in the terms collection
      */
     fluid.defaults("sjrk.storyTelling.transforms.arrayToString", {
         "gradeNames": [ "fluid.standardTransformFunction", "fluid.multiInputTransformFunction" ],
         "inputVariables": {
             "separator": ", ",
-            "stringOnly": "",
+            "trailingSeparator": false,
+            "stringOnly": true,
             "path": ""
         }
     });
 
     sjrk.storyTelling.transforms.arrayToString = function (input, extraInputs) {
-        var contentString = "";
-        var separator = extraInputs.separator();
-        var stringOnly = extraInputs.stringOnly();
-        var path = extraInputs.path();
+        var combinedString = "";
+        var separator = extraInputs.separator(),
+            trailingSeparator = extraInputs.trailingSeparator(),
+            stringOnly = extraInputs.stringOnly(),
+            path = extraInputs.path();
 
         fluid.each(input, function (term) {
-            if (path) {
-                term = fluid.get(term, path) || term;
-            }
+            term = fluid.get(term, path);
 
             if (!stringOnly || (term && typeof term === "string")) {
-                contentString += term + (separator || "");
+                combinedString += term + (separator || "");
             }
         });
 
-        return contentString.substring(0, contentString.lastIndexOf(separator));
+        if (!trailingSeparator) {
+            combinedString = combinedString.substring(0, combinedString.lastIndexOf(separator));
+        }
```

Comment:
It might be easier to just push acceptable strings into an array and calling the [join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) method.

Reply:
This is a good suggestion and I'll explore this a little more. The function was previously literally just a call to .join, but it has since become more complex.